diff --git a/dwm.c b/dwm.c
index 357d957..171895d 100644
--- a/dwm.c
+++ b/dwm.c
@@ -278,6 +278,7 @@ static void zoom(const Arg *arg);
 static Systray *systray = NULL;
 static const char broken[] = "broken";
 static char stext[256];
+static int sisurgent;
 static int screen;
 static int sw, sh;           /* X display screen geometry width, height */
 static int bh;               /* bar height */
@@ -873,7 +874,7 @@ drawbar(Monitor *m)
 
 	/* draw status first so it can be overdrawn by tags later */
 	if (m == selmon) { /* status is only drawn on selected monitor */
-		drw_setscheme(drw, scheme[SchemeNorm]);
+		drw_setscheme(drw, scheme[sisurgent ? SchemeUrg : SchemeNorm]);
 		tw = TEXTW(stext) - lrpad / 2 + 2; /* 2px extra right padding */
 		XSetErrorHandler(xerrordummy);
 		drw_text(drw, m->ww - tw - stw, 0, tw, bh, lrpad / 2 - 2, stext, 0);
@@ -2448,6 +2449,12 @@ updatestatus(void)
 {
 	if (!gettextprop(root, XA_WM_NAME, stext, sizeof(stext)))
 		strcpy(stext, "dwm-"VERSION);
+	if (strncmp(stext, "[URG]", 5) == 0) {
+		strcpy(stext, &stext[5]);
+		sisurgent = 1;
+	} else {
+		sisurgent = 0;
+	}
 	drawbar(selmon);
 	updatesystray();
 }
